### PR TITLE
handles onto errors better

### LINKED ISSUE #148: Handle http status codes for synonyms
Onto synonyms API client detects and handles errors based on if the content fails to json parse. 

https://github.com/helxplatform/dug/blob/43b14559e7ced9e065c3abe6a1e996c12e9e54e2/src/dug/core/annotate.py#L357

this should probably detect errors based on response http status codes .

On latest onto.renci.org instance a 400 hundred http status code is returned with a valid json. But this json is not an array. So this causes an exception up in the call chain when trying to add synonyms to actual data elements.

### PR DESCRIPTION
**Pull request by YaphetKG**
Resolves #148

### CODE REVIEW

**Comment by YaphetKG** [comment]
converting to draft to fix test failure

File: src/dug/core/annotate.py [hunk 1]
```diff
@@ -352,10 +352,17 @@ def make_request(self, curie: str, http_session: Session):
...
-            return response
+            response = http_session.get(url)
+            if response.status_code == 200:
+                return response.json()
+            if response.status_code == 400:
+                logger.error(f"No synonyms returned for: `{curie}`. Validation error.")
+                return []
+            if response.status_code == 500:
+                logger.error(f"No synonyms returned for: `{curie}`. Internal server error from {self.url}.")
+                return []
```
Comment (cschreep) [inline src/dug/core/annotate.py:363]: This needs a handler for the default case

File: src/dug/core/annotate.py [hunk 2]
```diff
@@ -352,10 +352,17 @@ def make_request(self, curie: str, http_session: Session):
...
         except json.decoder.JSONDecodeError as e:
-            logger.error(f"No synonyms returned for: {curie}")
+            logger.error(f"No synonyms returned for: `{curie}`. Json parse error.")
```
Comment (cschreep) [inline src/dug/core/annotate.py:365]: We should also log either the exception's message or the string that caused the parse error

**Review by cschreep — CHANGES_REQUESTED** [review]

File: src/dug/core/annotate.py (hunk 2 above)
Comment (YaphetKG) [reply src/dug/core/annotate.py:365]: maybe log the url that was used ?

**Review by cschreep — APPROVED** [review]
Looks good

**Pull request merged**